pkg/azuredisk/azuredisk_v2.go

```diff
@@ -405,11 +408,46 @@ func (d *DriverV2) StartControllersAndDieOnExit(ctx context.Context) {
 		klog.Errorf("Failed to initialize NodeAvailabilityController. Error: %v. Exiting application...", err)
 		os.Exit(1)
 	}
-	err = controller.WatchObject(mgr, source.Kind{Type: &v1.Namespace{}})
+	err = controller.WatchObject(mgr, source.Kind{Type: &v1.Namespace{}}, predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
+			return false
+		},
+		UpdateFunc: func(_ event.UpdateEvent) bool {
+			return true
+		},
+		GenericFunc: func(_ event.GenericEvent) bool {
+			return false
+		},
+		DeleteFunc: func(_ event.DeleteEvent) bool {
+			return false
+		},
```

Comment:
#1542 added the watcher to keep namespaces cached (I thought). Ignoring create and delete events would cause the cache to be stale. @sunpa93 Is this change correct?

Reply:
As far as I know, the predicates are strictly for event handler and wouldn't affect the caching behavior. So, we actually should be good if all predicates were to return false.

Reply:
I see now that this was the predicate originally used in the `controller.WatchObject` implementation for #1542. I'll assume this is correct, then.